.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,87 @@
+        "cmake.mergedCompileCommands": "${workspaceFolder}/compile_commands.json",
+        "cmakeExplorer.suiteDelimiter": ".",
+        "cmakeExplorer.debugConfig": "(gdb) Launch",
+        "editor.formatOnSave": true,
+        "[cpp]": {
+          "editor.defaultFormatter": "xaver.clang-format",
+          "editor.tabSize": 2
+        },
+        "clang-tidy.fixOnSave": true,
+        "github-enterprise.uri": "https://gitenterprise.xilinx.com",
+        "clang-tidy.executable": "clang-tidy-10",
+        "clangd.path": "/usr/bin/clangd-15",
+        "clangd.arguments": [
+          "--background-index",
+          "-header-insertion=never",
+          "-j=4"
+        ],
+        "python.formatting.provider": "black",
+        "cmake.buildDirectory": "${workspaceFolder}/build",
+        "cmake.generator": "Ninja",
+        "cmake.defaultVariants": {
+          "buildType": {
+            "default": "reldeb",
+            "description": "The build type.",
+            "choices": {
+              "debug": {
+                "short": "Debug",
+                "long": "Disable optimizations - include debug information.",
+                "buildType": "Debug"
+              },
+              "release": {
+                "short": "Release",
+                "long": "Optimize for speed - exclude debug information.",
+                "buildType": "Release"
+              },
+              "minsize": {
+                "short": "MinSizeRel",
+                "long": "Optimize for smallest binary size - exclude debug information.",
+                "buildType": "MinSizeRel"
+              },
+              "reldeb": {
+                "short": "RelWithDebInfo",
+                "long": "Optimize for speed - include debug information.",
+                "buildType": "RelWithDebInfo"
+              }
+            }
+          }
+        }
+      },
+      "extensions": [
+        "llvm-vs-code-extensions.vscode-clangd",
+        "vadimcn.vscode-lldb",
+        "ms-vscode.cmake-tools",
+        "fredericbonnet.cmake-test-adapter",
+        "twxs.cmake",
+        "xaver.clang-format",
+        "notskm.clang-tidy",
+        "ms-python.python",
+        "ms-python.vscode-pylance",
+        "github.vscode-pull-request-github",
+        "eamodio.gitlens",
+        "hediet.vscode-drawio",
+        "grapecity.gc-excelviewer",
+        "joaompinto.vscode-graphviz",
+        "mutantdino.resourcemonitor",
+        "llvm-vs-code-extensions.vscode-mlir",
+        "cschlosser.doxdocgen",
+        "chneau.resource-monitor",
+        "spmeesseman.vscode-taskexplorer",
+        "edonet.vscode-command-runner"
+      ]
+    }
+  },
+  "containerEnv": {
+    "CCACHE_MAXSIZE": "50G",
+    "CCACHE_NOHASHDIR": "true"
+  },
+  "workspaceFolder": "${localWorkspaceFolder}",
+  "workspaceMount": "source=${localWorkspaceFolder},target=${localWorkspaceFolder},type=bind,consistency=cached",
+  "onCreateCommand": "pip install commitizen black clang-format==14.0.6"
```

Comment:
You can also just add this to the docker file, but since you are removing the black settings and stuff you could remove this line here too.